core/src/main/java/org/arquillian/cube/impl/client/container/ProtocolMetadataUpdater.java

```diff
@@ -27,21 +28,35 @@ public void update(@Observes EventContext<ProtocolMetaData> eventContext, Contai
 
         Cube<?> cube = registry.getCube(container.getName());
         if(cube != null) {
-            Binding binding = cube.bindings();
-            String gatewayIp = cube.bindings().getIP();
+            HasMappedPorts mappedPorts = cube.getMetadata(HasMappedPorts.class);
+            Binding binding = cube.configuredBindings();
+            String gatewayIp = binding.getIP();
             for(Object contextObj : originalMetaData.getContexts()) {
                 if(contextObj instanceof HTTPContext) {
                     HTTPContext context = (HTTPContext)contextObj;
-                    PortBinding mapped = binding.getBindingForExposedPort(context.getPort());
                     String ip = context.getHost();
                     int port = context.getPort();
-                    if(mapped != null && port != mapped.getBindingPort()) {
+                    final String bindingIp;
+                    final Integer bindingPort;
+                    if (mappedPorts != null && mappedPorts.forTargetPort(port) != null) {
+                        bindingIp = mappedPorts.getIP();
+                        bindingPort = mappedPorts.forTargetPort(port);
+                    } else {
+                        PortBinding mapped = binding.getBindingForExposedPort(context.getPort());
```

Comment:
Could you add a Metadata impl for DockerCube as well that rely on Cube.binding() for now. Then we can at least focus on only one of the 'methods of getting binding data' in this part of the code.